src/pip/_internal/req/__init__.py

```diff
@@ -94,6 +85,49 @@ def install_given_reqs(
     return [i for i in installed if isinstance(i, InstallationResult)]
 
 
+def __safe_pool_map(
+        func,               # type: Callable
+        iterable,           # type: Iterable[Any]
+):
+    # type: (...) -> Optional[List[None, InstallationResult, BaseException]]
+    """
+    Safe call to Pool map, if Pool is not available return None
+
+    """
+    # Disable multiprocessing on Windows python 2.7
+    if sys.platform == 'win32' and sys.version_info.major == 2:
+        return None
+
+    if not iterable or Pool is None:
+        return None
```

Comment:
Definitely prettier :)
I was wondering on the behavior of the "func" itself, as in background mode, it will not throw exceptions directly... But then I realized that it will be consistent with the Pool behavior, so that's okay with me.

Reply:
Merged Pool context comment from @deveshks

Reply:
I wonder which exception is raised if pool is used on py27 on Windows though (side note: there's `PY2` from `pip._vendor.six` if you like it better), and if we can group it to the handling below.  Additionally IMHO `not iterable` should not be treated as a special case and if `Pool is None` we should also return `[func(i) for i in iterable]`.